TheMET/TheMET/Generic Types/LoadingStatus.swift

```diff
@@ -7,8 +7,14 @@
 
 import Foundation
 
-enum LoadingStatus<T> {
-    case loaded(T)
+enum LoadingStatus<LoadedData, FailedData> {
+    case loaded(LoadedData)
     case loading
-    case failed
+    case failed(FailedData)
+}
+
+
+enum FailedData {
```

Comment:
1) Let's not mix this enum with generic `LoadingStatus` in one file. Create a separate one.
2) Bad naming. It is not a `FailedData`, these are the reasons why the loading fails. `Loading error` suits it better.
3) You've created a single enum to be used in all possible loading cases. It creates invalid states. For example, loading go the image can't fill due to `no-searching results` failure. Pls create a separate enum for each particular context.